### PR TITLE
Upgrade mypy to 0.780

### PR DESCRIPTION
**Pull request by dbrgn**
Version 0.700 is not compatible with Python 3.8.

### CODE REVIEW

**Comment by dbrgn** [comment]
I think the tests fail because some test dependencies were not sufficiently pinned. From the pip install logs:

```
ERROR: pytest-asyncio 0.14.0 has requirement pytest>=5.4.0, but you'll have pytest 3.10.1 which is incompatible.
ERROR: pytest-cov 2.10.1 has requirement pytest>=4.6, but you'll have pytest 3.10.1 which is incompatible.
ERROR: pytest-mock 3.3.0 has requirement pytest>=5.0, but you'll have pytest 3.10.1 which is incompatible.
```

I'll fix the mypy errors.

File: saltyrtc/server/server.py [hunk 1]
```diff
@@ -951,7 +951,7 @@ def clean(self, path: Path) -> None:
 class Server:
     subprotocols = [
         SubProtocol.saltyrtc_v1.value
-    ]  # type: ClassVar[Sequence[SubProtocol]]
+    ]  # type: ClassVar[Sequence[str]]
```
Comment (dbrgn) [inline saltyrtc/server/server.py:954]: Note: I went for the quickfix here by changing the type to str. This could also be solved by changing all uses of `Server.subprotocols` to `[sp.value for sp in Server.subprotocols]`, but I don't know the codebase well and was a bit afraid to break something. If the only usage is on line 181, that would be an easy fix though.

**Comment by dbrgn** [comment]
Linting check still fails because the isort dependency wasn't properly pinned and changed behavior in the meantime. @lgrahl I leave it to you whether you want to update import sort order, or whether you also want to update the version specifier. (I'm a big fan of pinning dev dependencies, btw :stuck_out_tongue:)

File: saltyrtc/server/__init__.py [hunk 2]
```diff
@@ -13,16 +13,18 @@
 from .task import *  # noqa
 from .util import *  # noqa
 
+from . import common, events, exception, message, protocol, server, task, util
```
Comment (lgrahl) [inline saltyrtc/server/__init__.py:16]: Is this change needed to get mypy to run? Fairly certain there was a reason why I did not do this but it's been a long time.

File: saltyrtc/server/server.py (hunk 1 above)
Comment (lgrahl) [reply saltyrtc/server/server.py:954]: Maybe add a _todo_ comment above

File: saltyrtc/server/__init__.py (hunk 2 above)
Comment (dbrgn) [reply saltyrtc/server/__init__.py:16]: I think so, yes.

Comment (dbrgn) [reply saltyrtc/server/__init__.py:16]: Actually, it may be a mypy bug. Let me check...

File: saltyrtc/server/server.py (hunk 1 above)
Comment (threema-danilo) [reply saltyrtc/server/server.py:954]: Done in #123.

File: saltyrtc/server/__init__.py (hunk 2 above)
Comment (threema-danilo) [reply saltyrtc/server/__init__.py:16]: Reverted in #123. Let's hope tests pass, locally it seems to work.
